src/Serializer/Filter/PropertyFilter.php

```diff
@@ -129,18 +134,42 @@ private function getProperties(array $properties, array $whitelist = null): arra
 
         foreach ($properties as $key => $value) {
             if (is_numeric($key)) {
-                if (\in_array($value, $whitelist, true)) {
-                    $result[] = $value;
+                if (\in_array($propertyName = $this->denormalizePropertyName($value), $whitelist, true)) {
```

Comment:
note for self, need to check here that `$value` is a string if not done before so it would not blow with strict typing